mrp_production_partner_note/models/procurement_order.py

```diff
@@ -0,0 +1,19 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, api
+
+
+class ProcurementOrder(models.Model):
+    _inherit = 'procurement.order'
+
+    @api.model
+    def _prepare_mo_vals(self, procurement):
+        res = super(ProcurementOrder, self)._prepare_mo_vals(procurement)
+        sale_proc = procurement.move_dest_id.procurement_id
+        mrp_notes = sale_proc.sale_line_id.order_id.partner_id.mrp_notes
+        if mrp_notes:
+            res['notes'] = mrp_notes
```

Comment:
yea, but somebody can reuse your method and pass something else, the validation for sum is less explicit than the one for format.

```python
Python 2.7.6 (default, Sep  9 2014, 15:04:36)
[GCC 4.2.1 Compatible Apple LLVM 6.0 (clang-600.0.39)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 'text' + 1
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: cannot concatenate 'str' and 'int' objects
>>> 'text  {value}'.format(value=1)
'text  1'
>>>
```

It is more secure use the correct approach for strings than the + element at least that you want a hiden error in some moment.